python_modules/dagster/dagster_tests/core_tests/resource_tests/test_struct_config_resources.py

```diff
@@ -1781,3 +1782,33 @@ def hello_world_job() -> None:
 
     assert defs.get_job_def("hello_world_job").execute_in_process().success
     assert outputs == ["foo"]
+
+
+def test_aliased_field_structured_resource():
+    out_txt = []
+
+    class WriterResource(ConfigurableResource):
+        prefix_: str = PyField(..., alias="prefix")
+
+        def output(self, text: str) -> None:
+            out_txt.append(f"{self.prefix_}{text}")
+
+    @op
+    def hello_world_op(writer: WriterResource):
+        writer.output("hello, world!")
+
+    @job(resource_defs={"writer": WriterResource(prefix="")})
+    def no_prefix_job():
+        hello_world_op()
+
+    assert no_prefix_job.execute_in_process().success
+    assert out_txt == ["hello, world!"]
+
+    out_txt.clear()
+
+    @job(resource_defs={"writer": WriterResource(prefix="greeting: ")})
+    def prefix_job():
+        hello_world_op()
+
```

Comment:
Would be good to test runtime config too (e.g. `WriterResource.configure_at_launch()` and then pass the `{"prefix": "foo"}` to `execute_in_process`.